### PR TITLE
feat: rework Dockerfile and tag generation for dockerhub

### PR DESCRIPTION
**Pull request by daeMOn63**
Reworked a bit the docker entrypoint for public images, so it's easier to use.

- It does not depend anymore on external repo serving a bootstrap file, instead, the default settings for a given network are injected at build time by the CI, depending on the tag version, and the rest is extracted from the genesis at startup time.
- Introduced some *named* tags for our default networks: `fetchhub` `beaconworld` and `agentworld`.
- Updated docs with some sample commands for different usage
- Swapped CI to use new actions for building and tagging docker images

Sample tags from build output available at https://hub.docker.com/repository/docker/daemon63/fetchd

Also gave a try to update the docker hub repository description, but it's not working when using access tokens, it requires the plain password instead: https://github.com/peter-evans/dockerhub-description/issues/10. So we'll probably have to do without that (maybe just adding a link on docker hub to our documentation website ?)

Remaining TODOs:
- update and rebuild 0.2.x and 0.6.x
- cleanup tags on hub 0.3.x, 0.4.x...
- see if we have ways to automate docs update [./docs/docs/docker-images.md](./docs/docs/docker-images.md), but maybe we can just drop the links and reference general versions instead so we don't have to deal with it anymore.
- cleanup other Dockerfile targets, cleanup GCR image and entrypoints

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,48 +11,50 @@ RUN make install
...
-    apt-get install -y $PACKAGES
+    apt-get install -y jq curl
 
 COPY --from=builder /go/pkg/mod/github.com/\!cosm\!wasm/go-cosmwasm@v*/api/libgo_cosmwasm.so /usr/lib/libgo_cosmwasm.so
 COPY --from=builder /go/bin/fetchcli /usr/bin/fetchcli
 COPY --from=builder /go/bin/fetchd /usr/bin/fetchd
 COPY entrypoints/entrypoint.sh /usr/bin/entrypoint.sh
 
 VOLUME /root/.fetchd
-VOLUME /root/secret-temp-config
```
Comment (ejfitzgerald) [inline Dockerfile:31]: Can you double check that we no longer use this path in our developments @daeMOn63 . I seem to remember that it was added for this reason. Although I am very much in favour of removing this

File: docs/docs/networks.md [hunk 2]
```diff
@@ -50,6 +50,7 @@ The Agent World incentivized test network is phase 1 of our journey to Mainnet v
 | REST Endpoint  | https://rest-agentworld.fetch.ai:443                                         |
 | Block Explorer | [https://explore-agentworld.fetch.ai/](https://explore-agentworld.fetch.ai/) |
 | Token Faucet   | n/a                                                                          |
+| Seed Node(s)   | b387f0909e0eac780b8bbe25f04953136b61d0e4@connect-agentworld.prod.fetch-ai.com:26856 |
```
Comment (ejfitzgerald) [inline docs/docs/networks.md:53]: I would prefer not to advertise these links for validators on agent world. In a similar way to beaconworld I think that we need to wind this network down, move people across to the new test network and then progress from there.

File: .github/workflows/build_push.yml [hunk 3]
```diff
@@ -11,15 +11,64 @@ jobs:
...
+                LATEST=false
+            ;;
+            v0.2*)
+                NETWORK=agentworld
+                SEEDS="b387f0909e0eac780b8bbe25f04953136b61d0e4@connect-agentworld.prod.fetch-ai.com:26856"
+                RPC_ENDPOINT="https://rpc-agentworld.fetch.ai:443"
+                LATEST=false
+            ;;
+            *)
+                NETWORK=fetchhub
```
Comment (ejfitzgerald) [inline .github/workflows/build_push.yml:31]: I like this approach but I think that we need to think about a way to bake more that one network configuration into the image for a given reference or maybe it is simply a new tag of the image which would fit in this model in a more clean way.

In any case we need a way for users to be able to connect to fetchhub or testfetchhub network

File: Dockerfile (hunk 1 above)
Comment (daeMOn63) [reply Dockerfile:31]: it's only used on internal builds. The public version doesn't need that. I've restored those volumes definition in their matching target, but imo this is deployment-related data, not something the application needs to know about.

File: docs/docs/networks.md (hunk 2 above)
Comment (daeMOn63) [reply docs/docs/networks.md]: ack removed for now

File: .github/workflows/build_push.yml (hunk 3 above)
Comment (daeMOn63) [reply .github/workflows/build_push.yml:31]: Yep, that's the hard choice here, hard to tell with the current state of versions and networks. My idea here in the long run was to have default images ready to connect to mainnet out of the box, and provide specific tags for other networks with a matching builtin config. I think this logic here may evolve in a (version + default settings => tag name) kind of matrix so we'd have on docker hub:

```
fetchd:vA.B.C fetchd:latest (default mainnet)
    |_ fetchd:fetchhub (default mainnet)
    |_ fetchd:fetchhubtest (default testnet)
    |_ fetchd:another-testnet (default another-testnet)
fetchd:vD.E.F
    |_ fetchd:another-beta-testnet (default another-beta-testnet)
```

**Pull request closed without merging**